scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHandshaker.java

```diff
@@ -471,6 +471,7 @@ private void receivedServerHelloDone(ServerHelloDone message) throws HandshakeEx
 			generateKeys(premasterSecret);
 			break;
 		case PSK:
+			recordLayer.pauseRetransmission();
```

Comment:
>I would be afraid of stale handshake, because the retransmission may be stopped by a spoofed message (but I didn't go in to details, if this is really possible).

If you can spoof the message, you can send any messages and so completely stop the handshake by sending a malformed message or just sending an alert.